src/components/QueryStackTraceModal/QueryStackTraceModal.tsx

```diff
@@ -0,0 +1,48 @@
+import { Modal } from 'antd';
+import SyntaxHighlighter from '../SyntaxHighlighter';
+import { Button } from '@crate.io/crate-ui-components';
+
+export type QueryStackTraceModalProps = {
+  modalTitle: string;
+  visible?: boolean;
+  onClose?: () => void;
+  query: string;
+  queryError: string;
+};
+
+function QueryStackTraceModal({
+  modalTitle,
+  visible = false,
+  onClose,
+  query,
+  queryError,
+}: QueryStackTraceModalProps) {
+  const onOk = () => {
+    if (onClose) {
+      onClose();
+    }
+  };
+
+  return (
+    <Modal
+      title={modalTitle}
+      open={visible}
+      onOk={onOk}
+      width="80%"
+      footer={
+        <div>
+          <Button onClick={onClose}>OK</Button>
+        </div>
+      }
+    >
+      <div className="max-h-[400px] overflow-auto flex flex-col gap-4">
+        <SyntaxHighlighter language="sql" title="Failed Query Execution">
+          {query}
+        </SyntaxHighlighter>
+        <SyntaxHighlighter title="Stack Trace">{queryError}</SyntaxHighlighter>
```

Comment:
Does it figure out what language the snippet is in somehow?.. Wonder how it works.

Reply:
No, I think it's like a "default" style (without any language)